### ISSUE TITLE
add `noindex` option to `mzoon build`

### ISSUE THREAD

**Issue by databasedav** [MoonZoon/MoonZoon#154]
for example, i don't want my staging deployments to be indexed

**Comment by MartinKavik** [comment]
There is the method `Frontend::index_by_robots`. Is it what you meant? If so, please close the issue.
```rust
async fn frontend() -> Frontend {
    Frontend::new()
        .index_by_robots(false)
        .title("My app")
        ....
```
Implementation details:
https://github.com/MoonZoon/MoonZoon/blob/0279065c57a5637f464ccd1adb0ddd1f331de2c7/crates/moon/src/frontend.rs#L46

https://github.com/MoonZoon/MoonZoon/blob/0279065c57a5637f464ccd1adb0ddd1f331de2c7/crates/moon/src/frontend.rs#L87-L91

**Comment by databasedav** [comment]
yes this is close, i want this exposed as a configuration option to `mzoon build`, i'll try to open a pr for it

**Comment by databasedav** [comment]
@MartinKavik actually i don't think i can do this alone, how would i add a new env var for `mzoon build` `NOINDEX=1` and thread it through to `index_by_robots`? an env var is ideal so i can set it to avoid indexing my staging deployment

**Comment by MartinKavik** [comment]
I've just tested the code:
```rust
Frontend::new()
        .index_by_robots(not(matches!(std::option_env!("NOINDEX"), Some("1"))))
```
with the command
```
NOINDEX=1 makers mzoon start
```
and it seems to work without problem so I would recommend it as a workaround until I'll design & implement a better alternative.
